### PR TITLE
Bau/convert objects to records

### PR DESCRIPTION
**Pull request by BeckaL**
## What

Converts some entities to records to remove some boilerplate.

## How to review

1. Code Review commit by commit

### CODE REVIEW

**Review by alhcomer — APPROVED** [review]
LGTM

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/entity/AccountRecoveryResponse.java [hunk 1]
```diff
@@ -3,17 +3,5 @@
...
-    public AccountRecoveryResponse(boolean accountRecoveryPermitted) {
-        this.accountRecoveryPermitted = accountRecoveryPermitted;
-    }
-
-    public boolean getAccountRecoveryPermitted() {
-        return accountRecoveryPermitted;
-    }
-}
+public record AccountRecoveryResponse(
+        @SerializedName("accountRecoveryPermitted") @Expose boolean accountRecoveryPermitted) {}
```
Comment (mrwilson) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/entity/AccountRecoveryResponse.java:7]: As the field name is the same as the property name, we don't need the `@SerializedName` annotation.

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/entity/CheckUserExistsResponse.java [hunk 2]
```diff
@@ -6,58 +6,10 @@
...
-    public List<LockoutInformation> getLockoutInformation() {
-        return lockoutInformation;
-    }
-}
+public record CheckUserExistsResponse(
+        @SerializedName("email") @Expose String email,
+        @SerializedName("doesUserExist") @Expose boolean doesUserExist,
+        @SerializedName("mfaMethodType") @Expose MFAMethodType mfaMethodType,
+        @SerializedName("phoneNumberLastThree") @Expose String phoneNumberLastThree,
+        @SerializedName("lockoutInformation") @Expose
```
Comment (mrwilson) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/entity/CheckUserExistsResponse.java:14]: Same thing here as above

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/entity/LoginResponse.java [hunk 3]
```diff
@@ -5,75 +5,12 @@
...
-}
+public record LoginResponse(
+        @SerializedName("redactedPhoneNumber") @Expose String redactedPhoneNumber,
+        @SerializedName("mfaRequired") @Expose @Required boolean mfaRequired,
+        @SerializedName(value = "latestTermsAndConditionsAccepted") @Expose @Required
+                boolean latestTermsAndConditionsAccepted,
+        @SerializedName("mfaMethodType") @Expose @Required MFAMethodType mfaMethodType,
+        @SerializedName("mfaMethodVerified") @Expose @Required boolean mfaMethodVerified,
+        @SerializedName(value = "passwordChangeRequired") @Expose @Required
+                boolean passwordChangeRequired) {}
```
Comment (mrwilson) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/entity/LoginResponse.java:16]: And here too

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/entity/AccountRecoveryResponse.java (hunk 1 above)
Comment (BeckaL) [reply frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/entity/AccountRecoveryResponse.java:7]: We discussed - this actually is currently needed as without further code changes the default serialisation is snake case

**Pull request merged**